app/javascript/src/controller_branches.js

```diff
@@ -171,6 +177,26 @@ BranchesController.createBranch = function($node) {
   return branch;
 }
 
+BranchesController.addBranchCombinator = function(args) {
+  var branch = args[0];
+  if( !branch.index == 0 ) {
+    branch.$node.before("<p class=\"branch-or\">or</p>");
+  }
+}
+
+BranchesController.removeBranchCombinator = function(node) {
+    var $or = $(node).prev('.branch-or').first();
+    if($or) {
```

Comment:
The _if_ condition will always be true since jQuery returns either a populated or empty collection.
To achieve your check you can update it to:
```
if($or.length) {
}
```

A failed checked will have the value 0 (also resolves to false in JS) but if anything is found the value will be greater (so handled as though true). 

Since jQuery handles a lot of error handling for you, it should also be possible to just do this:
`$(node).prev('.branch-or').first().remove();`

Should nothing be found, because jQuery is returning an empty collection, the jQuery code will handle the lack found element gracefully and without complaint (nothing will be removed).